frontend_tests/node_tests/integration_bot_widget.js

```diff
@@ -0,0 +1,111 @@
+set_global('page_params', {
+    is_admin: false,
+    realm_users: [],
+});
+
+set_global('$', global.make_zjquery());
+
+set_global('channel', {
+});
+
+add_dependencies({
+    people: 'js/people.js',
+    stream_data: 'js/stream_data.js',
+});
+
+set_global('blueslip', {});
+
+var integration_bot_widget = require('js/integration_bot_widget.js');
+
+(function test_set_integration_bot_url() {
+    var success_callback;
+    var error_callback;
+    var posted;
+
+    var form_data = {
+        append: function (field, val) {
+            form_data[field] = val;
+        },
+    };
+
+    set_global('csrf_token', 'token-stub');
+    set_global('jQuery', {
+        each: function (lst, f) {
+            _.each(lst, function (v, k) {
+                f(k, v);
+            });
+        },
+    });
+
+    set_global('FormData', function () {
+        return form_data;
+    });
+
+    channel.post = function (req) {
+        posted = true;
+        assert.equal(req.url, '/json/bots');
+        success_callback = req.success;
+        error_callback = req.error;
+    };
+
+    var external_api_uri_subdomain = "https://chat.zulip.org/api/v1/external/";
+    var integration_url = "airbrake";
+    var bot_full_name = "airbrake";
+    var bot_short_name = "airbrake";
+    var stream_name = "Airbrake";
+
+    var airbrake = {
+        subscribed: false,
+        color: 'blue',
+        name: 'Airbrake',
+        stream_id: 1,
+        in_home_view: false,
+    };
+
+    // Add stream to sub
+    stream_data.add_sub(stream_name, airbrake);
+
+    var bot = {
+        bot_full_name: bot_full_name,
+        bot_short_name: bot_short_name,
+        stream_name: stream_name,
+        external_api_uri_subdomain: external_api_uri_subdomain,
+        integration_url: integration_url,
+    };
+
+    blueslip.error = function () {
+    };
+
+    // Main function to be tested
+    integration_bot_widget.set_integration_bot_url(bot);
+
+    // Assert that the above function goes to the desired url by channel.post
+    assert(posted);
+
+    var xhr = {
+        responseText: '{"api_key": "1234567", "email": "airbrake@zulip.com"}',
+    };
+
+    // Call on_success callback function.
+    success_callback(xhr);
```

Comment:
Can we do a test for whether the bot email was subscribed to stream?
This is done as part of the `on_success()` so maybe we should test it as well.